coredhcp.go

```diff
@@ -136,6 +141,16 @@ func (s *Server) MainHandler6(conn net.PacketConn, peer net.Addr, req dhcpv6.DHC
 		}
 	}
 
+	// if the request was relayed, re-encapsulate the response
+	if req.IsRelay() {
+		tmp, err := dhcpv6.NewRelayReplFromRelayForw(req.(*dhcpv6.RelayMessage), resp.(*dhcpv6.Message))
```

Comment:
Is the `tmp` really needed?

Reply:
to avoid overwriting `resp`, yes, is that what you mean?

Reply:
Is there any problem with overwriting it?